### PR TITLE
[Wallet] Trivial cleanup of HD wallet changes

### PR DESCRIPTION
**Pull request by jonasschnelli**
Trivial cleanup.
Reported by @MarcoFalke and @dooglus.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
I think you can use __func__ because the wallet seems a place where a lot
refactoring is happening. So we don't have yo change it each time.

Anyway, concept ACK

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ CPubKey CWallet::GenerateNewHDMasterKey()
 
         // write the key&metadata to the database
         if (!AddKeyPubKey(key, pubkey))
-            throw std::runtime_error("CWallet::GenerateNewKey(): AddKey failed");
+            throw std::runtime_error("CWallet::GenerateNewHDMasterKey(): AddKey failed");
```
Comment (paveljanik) [inline src/wallet/wallet.cpp]: `AddKeyPubKey` failed, not `AddKey`. And it can't "fail" at all right now 8)

Comment (MarcoFalke) [inline src/wallet/wallet.cpp]: The goal was to keep this in case some refactoring makes it possible that it fails.

**Comment by jonasschnelli** [comment]
Fixed nits.

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -1200,7 +1200,7 @@ CPubKey CWallet::GenerateNewHDMasterKey()
 
         // write the key&metadata to the database
         if (!AddKeyPubKey(key, pubkey))
-            throw std::runtime_error("CWallet::GenerateNewKey(): AddKey failed");
+            throw std::runtime_error("CWallet::"+std::string(__func__)+" AddKeyPubKey failed");
```
Comment (paveljanik) [inline src/wallet/wallet.cpp]: : missing?

**Comment by jonasschnelli** [comment]
Fixed @paveljanik's nit.

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/8443/commits/a575426b8b5697f94fceb429de1e0019ddec0ecf

**Comment by MarcoFalke** [comment]
utACK a575426

**Comment by MarcoFalke** [comment]
@laanwj Anything holding this back?

File: src/wallet/wallet.cpp [hunk 3]
```diff
@@ -1200,7 +1200,7 @@ CPubKey CWallet::GenerateNewHDMasterKey()
 
         // write the key&metadata to the database
         if (!AddKeyPubKey(key, pubkey))
-            throw std::runtime_error("CWallet::GenerateNewKey(): AddKey failed");
+            throw std::runtime_error("CWallet::"+std::string(__func__)+": AddKeyPubKey failed");
```
Comment (laanwj) [inline src/wallet/wallet.cpp]: You shouldn't prefix CWallet:: here

Comment (jonasschnelli) [inline src/wallet/wallet.cpp]: Removed the prefix.

**Comment by MarcoFalke** [comment]
utACK 7e5d94d

**Pull request merged**